tools/data-mapper/Microsoft.Health.Tools.DataMapper/ClientApp/src/components/NavMenu.tsx

```diff
@@ -4,17 +4,60 @@
 // -------------------------------------------------------------------------------------------------
 
 import * as React from 'react';
-import { Navbar, NavbarBrand } from 'reactstrap';
-import { Link } from 'react-router-dom';
+import { Navbar, NavbarBrand, NavbarText } from 'reactstrap';
+import { Link, useLocation } from 'react-router-dom';
+import PersistService from '../services/PersistService';
+import { Mapping } from '../store/Mapping';
+import { default as MappingNameModal, Action } from './mapping/List.Modals.Name';
 import './NavMenu.css';
 
 const NavMenu = (props: {}) => {
+    const [mappingId, setMappingId] = React.useState('');
+    const [mappingTypeName, setMappingTypeName] = React.useState('');
+
+    const location = useLocation();
+
+    React.useEffect(() => {
+        const pathname = location.pathname;
+        const id = pathname.substr(pathname.lastIndexOf('/') + 1);
```

Comment:
Nit: please try a few invalid url path and see whether this can handle those situations properly.

Reply:
Thanks, @ms-teli. With your comment I made another commit to get the mapping ID from the URL path more reliably - especially if the path changes in the future.

Reply:
thanks for making the change.